plugin/fs/plugin.go

```diff
@@ -282,7 +282,23 @@ func (p FSPlugin) Store(endpoint plugin.ShieldEndpoint) (string, int64, error) {
 }
 
 func (p FSPlugin) Retrieve(endpoint plugin.ShieldEndpoint, file string) error {
-	return plugin.UNIMPLEMENTED
+	cfg, err := getFSConfig(endpoint)
+	if err != nil {
+		return err
+	}
+
+	f, err := os.Open(fmt.Sprintf("%s/%s", cfg.BasePath, file))
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	if _, err = io.Copy(os.Stdout, f); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
you can just `return io.Copy(os.Stdout, f)`

Reply:
By doing this, aren't we playing hot-potato with the error?